src/core/mempool.rs

```diff
@@ -750,6 +756,128 @@ impl MemPoolTxInfo {
     }
 }
 
+/// Used to locally cache nonces to avoid repeatedly looking them up in the nonce.
+struct NonceCache {
+    cache: HashMap<StacksAddress, u64>,
+    size: usize,
```

Comment:
why is `size` separate? getting the size of a `HashMap` should be cheap

Reply:
Yeah, good point. I think that is some artifact of this struct getting simplified, or my lack of trust for the implementation 😄 (I should know better with Rust).

Reply:
Removed in 56d0050d2

Reply:
I'm not sure what happened to that earlier commit. It is removed again in 5cd1c6194